### PR TITLE
Documentation: Improve/trim README

### PR DESCRIPTION
**Pull request by amotl**
Hi.

Similar to https://github.com/crate/crate-jdbc/pull/365, this patch aims to clarify the purpose of this repository. In this case, it significantly trims the README, to make the repository appear different than the upstream repository https://github.com/pgjdbc/pgjdbc.

My proposal for the new README can be best viewed here: [PostgreSQL JDBC Driver with adjustments for CrateDB](https://github.com/crate/pgjdbc/blob/f54cde77b61/README.md).

With kind regards,
Andreas.

### CODE REVIEW

File: README.md
Comment (amotl) [inline README.md:8]: Other than

> ... to support specific details for CrateDB, which can be inspected ...

it would be sweet to be able to add a more high-level summary here, similar to the "What's inside" section at [^1].

[^1]: https://github.com/crate/crate-jdbc/blob/159010bb29e3da5c90819e081167942a66a8ed2b/docs/index.rst#whats-inside

Comment (matriv) [reply README.md:8]: I think a simple sentence like, to avoid certain statements and queries on internal PostgreSQL tables which `were` (or are? I don't know) not supported by CrateDB.

Comment (amotl) [reply README.md]: Thanks, I've just updated the patch accordingly.

**Review by matriv — APPROVED** [review]
LGTM, please wait for a second review though.

**Pull request merged**
